resolwe/flow/expression_engines/jinja/filters.py

```diff
@@ -23,16 +23,22 @@ def _get_data_attr(data, attr):
 
 def name(data):
     """Return `name` of `Data`."""
+    if isinstance(data, (list, tuple)):
```

Comment:
Move the whole logic to a helper function, so you don't need to repeat it in each function.

Reply:
I agree with @dblenkus as it is repeating quite a lot here. Perhaps have something like `apply_filter_list(value, lambda item: _get_data_attr(item, 'name'))` and then you can handle both list and non-list cases in a generic fashion.